terraform/utils.go

```diff
@@ -0,0 +1,61 @@
+package terraform
+
+import (
+	"sort"
+	"strings"
+)
+
+func isValidVariableFile(fileName string) bool {
+	if strings.HasSuffix(fileName, DEFAULT_TFVARS) {
+		return true
+	}
+	for _, fileExt := range VALID_VARIABLE_FILES {
+		if strings.HasSuffix(fileName, fileExt) {
+			return true
+		}
+	}
+	return false
+}
+
+func isValidTerraformFile(fileName string) bool {
+	for _, fileExt := range VALID_TERRAFORM_FILES {
+		if strings.HasSuffix(fileName, fileExt) {
+			return true
+		}
+	}
+	return false
+}
+
+func orderFilesByPriority(fileNames []string) []string {
+	prioritisedFileNames := make([]string, 0, len(fileNames))
+
+	// Default values have lowest priority (in .tf files)
+	var defaultTfvars = ""
+	for _, fileName := range fileNames {
+		if strings.HasSuffix(fileName, TF) {
+			prioritisedFileNames = append(prioritisedFileNames, fileName)
+		} else if strings.HasSuffix(fileName, DEFAULT_TFVARS) {
```

Comment:
I think we should check `fileName == DEFAULT_TFVARS` to make sure we are not saving files named `test_terraform.tfvars` for example as the default tfvars file.

Reply:
Hmm good catch! That was what I had originally but the `fileName` can be something like `/path/to/terraform.tfvars`. I'll think of something else